### ISSUE TITLE
Add support to Fastify HTTP server

### ISSUE THREAD

**Issue by rsaz** [expressots/expressots#239]
## Description
Integrating Fastify support into your existing application structure, which currently supports primarily to Express, involves several key steps. The goal is to abstract common functionalities between Express and Fastify (or any future web servers we plan to support) so that our core application logic remains agnostic of the underlying web server technology.

## Requirements
- Abstract common interfaces between express and fastify, example: request handler, middleware (plugin), providers, initialization and server configuration
- Update AppFactory to accomodate any server of type `ServerAdapater`
- Middleware compatibility
- Adapt the `App` Class
- Route handling
- Application lifecycle
- Inversify decorators and container

**Comment by shravan20** [comment]
@rsaz : Can I pick this up?

**Comment by rsaz** [comment]
> [@rsaz](https://github.com/rsaz) : Can I pick this up?

Yes you can. I will assign it to you. Have in mind that the task is big. Here are the requirements

- Create a repo using our [template](https://github.com/expressots/expressots-project-template)
- Name the project `adapter-fastify`
- Fastify supports expressjs like middlewares but also plugin pattern. For the middleware side we already have support on `core`
- Either generalize the decorators to work with Fastify or create new ones that matches the current decorators signatures for this new adapter
- The new adapter needs to follow a similar structure of the `adapter-express` as we currently provide App Factory that is able to support any web server as long as they implement `IWebServer` interface

Following these steps you should be successful.

**Comment by rsaz** [comment]
Superseded by #940 — good news: the abstraction this issue asked for (IWebServer, adapter-agnostic AppFactory) shipped in the 4.x monorepo. Only the Fastify adapter itself remains, and #940 breaks it into mergeable milestones.
